code/modules/mapping/mapping_helpers.dm

```diff
@@ -1359,3 +1359,35 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 
 	engraved_wall.AddComponent(/datum/component/engraved, engraving["story"], FALSE, engraving["story_value"])
 	qdel(src)
+
+/obj/effect/mapping_helpers/emergency_shuttle_event_override
+	name = "emergency shuttle events override"
+	icon_state = "shuttleeventoverride"
+	late = TRUE
+	/// List of event paths to run
+	var/list/events_override = list()
+
+/obj/effect/mapping_helpers/emergency_shuttle_event_override/LateInitialize()
+	var/obj/docking_port/mobile/port = SSshuttle.emergency
+	if(!port)
```

Comment:
For future reference, LateInitialize needs a parent call (but you should use the template's `post_load`